src/CSharp/CodeCracker/Usage/DisposableVariableNotDisposedAnalyzer.cs

```diff
@@ -38,6 +38,8 @@ private static void AnalyzeObjectCreation(SyntaxNodeAnalysisContext context)
             if (objectCreation == null) return;
             if (objectCreation?.Parent is UsingStatementSyntax) return;
             if (objectCreation?.Parent is ReturnStatementSyntax) return;
+            if (objectCreation.FirstAncestorOfType<ConstructorInitializerSyntax>()?.Kind() == SyntaxKind.ThisConstructorInitializer) return;
+            if (objectCreation.FirstAncestorOfType<ObjectCreationExpressionSyntax>() != null) return;
```

Comment:
@thorgeirk11 Reflection is always costly. But this type of code appears everywhere, on many analyzers. We are not measuring those yet, so I wouldn't worry at this moment. What I would recommend is that you check other faster things first, and this is already happening.
BTW, the two calls above code be faster using `IsKind`. :)